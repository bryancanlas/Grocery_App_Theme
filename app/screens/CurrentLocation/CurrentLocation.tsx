import React, { useState } from 'react'
import { StyleSheet, Linking, Image, View as DefaultView } from 'react-native'
import { Colors, moderateScale } from '../../constants'
import { View, Text, TouchableOpacity } from '../../components/Themed'
import * as Location from "expo-location";
import { FlashMessage } from "../../components/FlashMessage";
import Spinner from "../../components/Spinner";

export default function CurrentLocation(props: any) {
  const [loading, setLoading] = useState(false);

  const getCurrentLocation = async () => {
    setLoading(true)
    console.log('this render')
    let { status, canAskAgain } = await Location.requestForegroundPermissionsAsync();
    if (status !== 'granted') {
      setLoading(false);
      FlashMessage({
        message:
          'Tap on this message to open Settings then allow app to use location from permissions.',
        onPress: async () => {
          await Linking.openSettings()
        }
      })
      return;
    }
    try{
      let location = await Location.getCurrentPositionAsync({});
      setLoading(false)
      props.navigation.replace('Home')
    }
    catch(err){
      alert(err)
      setLoading(false)
    }
    

  }

  return (

    <View testID={'firstView'} style={styles.container}>
      {
        loading
        &&
        <DefaultView style={{ top: '67%', position: 'absolute', zIndex: 1 }}>
          <Spinner />
        </DefaultView>
      }
      <View testID={'SecondView'} style={styles.fullWidth}>
        <Image style={styles.imageStyle} source={{ uri: 'https://picsum.photos/200/300' }} />
      </View>
      <View style={styles.fullWidth}>
        <View style={styles.marginBottom}>
          <Text testID={'helperText'} style={styles.titleText}>grocery uses your location to show the resturant near you!</Text>
        </View>
        <TouchableOpacity disabled={loading} onPress={getCurrentLocation} style={[styles.buttonStyle, { backgroundColor: loading ? Colors.light.placeholderText : Colors.light.primary }]} lightColor={Colors.light.primary}>
          <Text style={styles.titleText} lightColor={Colors.light.background}>Use Current Location</Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity disabled={loading} activeOpacity={0.7} onPress={() => { props.navigation.navigate('SelectLocation') }}>
        <Text style={styles.titleText} lightColor={Colors.light.primary}>
          {'Select another location'}
        </Text>
      </TouchableOpacity>

    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    padding: moderateScale(20),
    justifyContent: 'flex-end'
  },
  titleText: {
    fontFamily: 'OpenSans-Bold',
    fontSize: moderateScale(15),
    textAlign: 'center'
  },
  imageStyle: {
    width: moderateScale(200),
    height: moderateScale(250)
  },
  buttonStyle: {
    justifyContent: 'flex-end',
    padding: moderateScale(15),
    borderRadius: moderateScale(10),
    alignItems: 'center',
    width: '100%'
  },
  fullWidth: {
    width: '100%',
    marginBottom: moderateScale(80),
    alignItems: 'center',
  },
  marginBottom: {
    marginBottom: moderateScale(20)
  },

})