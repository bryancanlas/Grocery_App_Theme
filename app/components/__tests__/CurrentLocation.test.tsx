import React from "react";
import renderer from "react-test-renderer";
import CurrentLocation from "../../screens/CurrentLocation";

let findElement = function(tree:any, Element:string)
{
    let node;
        // if(tree.children[node].props.testID == Element)
        // {
        //     return true
        // }
    if(tree.children[1].children[0].children[0].props.testID == Element)
    return true

    return undefined
}

// it('find element', () => {
//     const tree = renderer.create(<CurrentLocation />).toJSON();
//     expect(findElement(tree,'username')).toBeDefined();
//   });

it('should render correctly', ()=>{
    let tree = renderer.create(<CurrentLocation />).toJSON()
    // expect(findElement(tree,'helperText')).toBeDefined()
    expect(tree).toMatchSnapshot();
})