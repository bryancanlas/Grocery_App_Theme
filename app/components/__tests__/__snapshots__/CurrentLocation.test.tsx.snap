// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create item 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "shadowColor": "#000",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "padding": 31.428571428571427,
      },
    ]
  }
  testID="firstView"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e0e0e0",
          "shadowColor": "#000",
        },
        Object {
          "alignItems": "center",
          "marginBottom": 125.71428571428571,
          "width": "100%",
        },
      ]
    }
    testID="SecondView"
  >
    <Image
      source={
        Object {
          "uri": "https://picsum.photos/200/300",
        }
      }
      style={
        Object {
          "height": 392.85714285714283,
          "width": 314.2857142857143,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e0e0e0",
          "shadowColor": "#000",
        },
        Object {
          "alignItems": "center",
          "marginBottom": 125.71428571428571,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#e0e0e0",
            "shadowColor": "#000",
          },
          Object {
            "marginBottom": 31.428571428571427,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontFamily": "OpenSans-Bold",
              "fontSize": 23.57142857142857,
              "textAlign": "center",
            },
          ]
        }
        testID="helperText"
      >
        grocery uses your location to show the resturant near you!
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D70F64",
          "borderRadius": 15.714285714285714,
          "justifyContent": "flex-end",
          "opacity": 1,
          "padding": 23.57142857142857,
          "shadowColor": "#000",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontFamily": "OpenSans-Bold",
              "fontSize": 23.57142857142857,
              "textAlign": "center",
            },
          ]
        }
      >
        Use Current Location
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "opacity": 1,
        "shadowColor": "#000",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#D70F64",
          },
          Object {
            "fontFamily": "OpenSans-Bold",
            "fontSize": 23.57142857142857,
            "textAlign": "center",
          },
        ]
      }
    >
      Select another location
    </Text>
  </View>
</View>
`;

exports[`should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "shadowColor": "#000",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "padding": 31.428571428571427,
      },
    ]
  }
  testID="firstView"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e0e0e0",
          "shadowColor": "#000",
        },
        Object {
          "alignItems": "center",
          "marginBottom": 125.71428571428571,
          "width": "100%",
        },
      ]
    }
    testID="SecondView"
  >
    <Image
      source={
        Object {
          "uri": "https://picsum.photos/200/300",
        }
      }
      style={
        Object {
          "height": 392.85714285714283,
          "width": 314.2857142857143,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e0e0e0",
          "shadowColor": "#000",
        },
        Object {
          "alignItems": "center",
          "marginBottom": 125.71428571428571,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#e0e0e0",
            "shadowColor": "#000",
          },
          Object {
            "marginBottom": 31.428571428571427,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontFamily": "OpenSans-Bold",
              "fontSize": 23.57142857142857,
              "textAlign": "center",
            },
          ]
        }
        testID="helperText"
      >
        grocery uses your location to show the resturant near you!
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D70F64",
          "borderRadius": 15.714285714285714,
          "justifyContent": "flex-end",
          "opacity": 1,
          "padding": 23.57142857142857,
          "shadowColor": "#000",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontFamily": "OpenSans-Bold",
              "fontSize": 23.57142857142857,
              "textAlign": "center",
            },
          ]
        }
      >
        Use Current Location
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "opacity": 1,
        "shadowColor": "#000",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#D70F64",
          },
          Object {
            "fontFamily": "OpenSans-Bold",
            "fontSize": 23.57142857142857,
            "textAlign": "center",
          },
        ]
      }
    >
      Select another location
    </Text>
  </View>
</View>
`;
